src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -780,6 +780,47 @@ public async void M()
             return CSharpVerifyExpectedCodeFixDiagnosticsAsync(originalSource, fixedSource, GetCSharpResult(12, 74, 12, 254));
         }
 
+
+        [Fact]
+        public Task CS_Fixer_PreserveNullability()
+        {
+            string originalSource = @"
+#nullable enable
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+public class C
+{
+    async void M(FileStream? stream, byte[]? buffer)
+    {
+        await stream!.ReadAsync(buffer!, offset: 0, count: buffer!.Length).ConfigureAwait(false);
+    }
+}
+";
+            string fixedSource = @"
+#nullable enable
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+public class C
+{
+    async void M(FileStream? stream, byte[]? buffer)
+    {
+        await (stream!).ReadAsync((buffer!).AsMemory(start: 0, length: buffer!.Length)).ConfigureAwait(false);
```

Comment:
`generator.MemberAccessExpression` internally adds the parenthesis. I couldn't find a way to exclude them. Seems to only happen when it detects a bang.

Reply:
Could you open an issue on roslyn? 

It's not ideal but I would not go for the complexity of doing a manual workaround here.

Reply:
Done: https://github.com/dotnet/roslyn/issues/50260